client/src/main/java/org/eclipse/hono/client/impl/HonoClientImpl.java

```diff
@@ -361,23 +367,23 @@ private void reconnect(
      * {@inheritDoc}
      */
     @Override
-    public Future<MessageSender> getOrCreateTelemetrySender(final String tenantId) {
+    public final Future<MessageSender> getOrCreateTelemetrySender(final String tenantId) {
         return getOrCreateTelemetrySender(tenantId, null);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public Future<MessageSender> getOrCreateTelemetrySender(final String tenantId, final String deviceId) {
+    public final Future<MessageSender> getOrCreateTelemetrySender(final String tenantId, final String deviceId) {
 
         Objects.requireNonNull(tenantId);
         return getOrCreateSender(
                 TelemetrySenderImpl.getTargetAddress(tenantId, deviceId),
                 () -> createTelemetrySender(tenantId, deviceId));
     }
 
-    private Future<MessageSender> createTelemetrySender(
+    private final Future<MessageSender> createTelemetrySender(
```

Comment:
private methods cannot be overridden, no need to declare them *final*

Reply:
oops. thats was not supposed to be there. will correct it.